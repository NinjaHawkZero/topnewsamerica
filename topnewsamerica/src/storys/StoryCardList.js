import React from "react";
import StoryCard from "./StoryCard";

//Renders list of story card


function StoryCardList({stories}) {

    
    return (
        <div class="col d-flex justify-content-center">

<div className="StoryCardList">
             <h1 className="font-weight-bold">Top News America</h1>
            <p className="lead">All of America's top stories!</p>

            {stories.map(story => (
                <StoryCard
                key={story.url}
                author={story.author}
                title={story.title}
                description={story.description}
                url={story.url}
                publishedAt={story.publishedAt}
                urlToImage={story.urlToImage}

                />
            ))}

        </div>
        </div>
       
    )

}

export default StoryCardList;