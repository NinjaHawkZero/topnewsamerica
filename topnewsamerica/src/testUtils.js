import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  password: "testpass",
  
};

const save = async function saveStory(story) {
  try{let res = await TopNewsApi.saveStory(story, currentUser.username);

  return {success: true}}
  catch(err) {
    console.error("Could not save story", err);
    return {success: false, err}
  }
}

const UserProvider =
    ({ currentUser = demoUser, saveStory = save}) => (
    <UserContext.Provider value={{ currentUser, saveStory }}>
      
    </UserContext.Provider>
);

export { UserProvider };