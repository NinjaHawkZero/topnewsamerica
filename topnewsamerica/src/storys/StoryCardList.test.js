import React from "react";
import { render } from "@testing-library/react";
import StoryCardList from "./StoryCardList";
import { UserProvider } from "../testUtils";
import { MemoryRouter } from "react-router-dom";


let stories = [{saved_by: 2, author:"Micheal John", title:"War in the middleeast", description:"There are conflicting forces in the middleeast", published_at:"2022-01-02T00:15:09Z", url:"https://www.macrumors.com/2022/01/01/apple-watch-life-saving-911-ad/", urlToImage:"https://images.macrumors.com/t/PvHWwlMR5LrYla3IjTpwcep860o=/1600x/article-new/2022/01/apple-watch-911-ad.jpeg"}]

it("matches snapshot with stories", function() {
    const { asFragment } = render(
    
 
    <MemoryRouter>
<UserProvider>
      <StoryCardList  stories={stories}/>
    </UserProvider>
    </MemoryRouter>
    
    
    );
    expect(asFragment()).toMatchSnapshot();
  });