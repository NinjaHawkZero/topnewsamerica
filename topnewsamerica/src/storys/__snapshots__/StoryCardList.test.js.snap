// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with stories 1`] = `<DocumentFragment />`;
